controller/work_item_link_category_blackbox_test.go

```diff
@@ -142,12 +146,91 @@ func (s *workItemLinkCategorySuite) createWorkItemLinkCategoryUser() (http.Respo
 	return test.CreateWorkItemLinkCategoryCreated(s.T(), s.svc.Context, s.svc, s.linkCatCtrl, &payload)
 }
 
+func createWorkItemLinkCategorySystemInRepo(t *testing.T, db application.DB, ctx context.Context) uuid.UUID {
+	id := "0e671e36-871b-43a6-9166-0c4bd573e231"
+	return createWorkItemLinkCategoryInRepo(t, db, ctx, "test-system", "This work item link category is reserved for the core system.", &id)
+}
+
+func createWorkItemLinkCategoryUserInRepo(t *testing.T, db application.DB, ctx context.Context) uuid.UUID {
+	id := "bf30167a-9446-42de-82be-6b3815152051"
+	return createWorkItemLinkCategoryInRepo(t, db, ctx, "test-user", "This work item link category is managed by an admin user.", &id)
+}
+
+func createWorkItemLinkCategoryInRepo(t *testing.T, db application.DB, ctx context.Context, name string, description string, id *string) uuid.UUID {
+	var cID uuid.UUID
+	if id == nil {
```

Comment:
This might be a personal preference but I hate passing `nil`. 

Please change the function signature from 
```go
func createWorkItemLinkCategoryInRepo(
t *testing.T, 
db application.DB, 
ctx context.Context, 
name string, 
description string, 
id *string) uuid.UUID {
```
to
```go
func createWorkItemLinkCategoryInRepo(
t *testing.T, 
db application.DB, 
ctx context.Context, 
linkCat link.WorkItemLinkCategory) uuid.UUID {
```
Then the caller absolutely knows what needs to be passed and if the category struct suddenly requires a new field, we don't have to modify all calls to this function.